src/middlewared/middlewared/pytest/unit/utils/test_filter_list.py

```diff
@@ -440,3 +472,34 @@ def test__filter_list_select_as_validation():
         # wrong type in select
         filter_list(DATA_SELECT_COMPLEX, [], {'select': [[1, 'cat']]})
         assert 'first item must be a string' in str(ve)
+
+
+def test__filter_list_timestamp():
+    # Test variations of not being ISO-8601 timestamp
+
+    # check for bogus string
+    with pytest.raises(ValueError) as ve:
+        filter_list([], [["timestamp.$date", "=", "Canary"]])
+
+    assert 'must be an ISO-8601 formatted timestamp string' in str(ve)
```

Comment:
All these assertions and `with pytest.raises` must be separate test cases so that if one fails, others are still tested